### PR TITLE
fix(net): set correct max size for Datagram

### PR DESCRIPTION
**Pull request by billywhizz**
Fixes [#21609](https://github.com/denoland/deno/issues/21609)

### CODE REVIEW

File: ext/net/01_net.js [hunk 1]
```diff
@@ -385,6 +387,11 @@ class Datagram {
   async send(p, opts) {
     switch (this.addr.transport) {
       case "udp":
+        if (p.length > UDP_DGRAM_MAXSIZE) {
```
Comment (billywhizz) [inline ext/net/01_net.js:390]: we could just take this check out and leave the PR with the change above. if we try to send with a payload greater than whatever the OS allows (it seems macs define a much lower max size by default we will throw back a system error. i can modify the test to handle this correctly.

File: ext/net/01_net.js [hunk 2]
```diff
@@ -11,6 +11,8 @@ import {
 import * as abortSignal from "ext:deno_web/03_abort_signal.js";
 import { SymbolDispose } from "ext:deno_web/00_infra.js";
 
+const UDP_DGRAM_MAXSIZE = 65507;
```
Comment (littledivy) [inline ext/net/01_net.js:14]: Should this be `65535`? (Total Length field of a IP packet header is 16-bits)

Comment (littledivy) [reply ext/net/01_net.js:14]: Ah sorry this is only the payload size. 65507 makes sense

File: cli/tests/unit/net_test.ts [hunk 3]
```diff
@@ -385,6 +385,70 @@ Deno.test(
...
+    for (const size of sizes) {
+      const tosend = new Uint8Array(size);
+      let byteLength = 0;
+      try {
+        byteLength = await alice.send(tosend, bob.addr);
+      } catch (err) {
+        // Note: we have to do the test this way as different OS's have
+        // different UDP size limits enabled, so we will just ensure if
+        // an error is thrown it is the one we are expecting.
+        assert(err.message.match(rx));
```
Comment (littledivy) [inline cli/tests/unit/net_test.ts:435]: This test is failing on Windows on this line. cc @billywhizz

Comment (billywhizz) [reply cli/tests/unit/net_test.ts:436]: yeah. every OS has a different error message seemingly. i'll see if i can fix for windows.

Comment (billywhizz) [reply cli/tests/unit/net_test.ts:435]: i fixed this but now seeing a failure here on linux which wasn't happening before. are these just flaky tests?

https://github.com/denoland/deno/actions/runs/7357751549/job/20029885353?pr=21611#step:42:3062

Comment (littledivy) [reply cli/tests/unit/net_test.ts:435]: I'm seeing that in another PR too. Looks like a flake yeah

Comment (billywhizz) [reply cli/tests/unit/net_test.ts:435]: i'm seeing a SERVFAIL status when i try to resolve that address locally using dig. so, possibly it is an intermittent issue on google dns side?

```shell
dig 8.8.8.8.in-addr.arpa

; <<>> DiG 9.18.18-0ubuntu0.22.04.1-Ubuntu <<>> 8.8.8.8.in-addr.arpa
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 7815
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;8.8.8.8.in-addr.arpa.          IN      A

;; Query time: 196 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Fri Dec 29 16:12:29 GMT 2023
;; MSG SIZE  rcvd: 49
```

i also see this when i query google 8.8.8.8 directly. :thinking: 

```shell
dig @8.8.8.8 8.8.8.8.in-addr.arpa

; <<>> DiG 9.18.18-0ubuntu0.22.04.1-Ubuntu <<>> @8.8.8.8 8.8.8.8.in-addr.arpa
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 35308
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
; EDE: 0 (Other): ([209.244.0.1] Lame delegation at 8.8.8.in-addr.arpa for 8.8.8.8.in-addr.arpa/a)
; EDE: 0 (Other): ([209.244.0.2] Lame delegation at 8.8.8.in-addr.arpa for 8.8.8.8.in-addr.arpa/a)
; EDE: 22 (No Reachable Authority): (At delegation 8.8.8.in-addr.arpa for 8.8.8.8.in-addr.arpa/a)
;; QUESTION SECTION:
;8.8.8.8.in-addr.arpa.          IN      A

;; Query time: 56 msec
;; SERVER: 8.8.8.8#53(8.8.8.8) (UDP)
;; WHEN: Fri Dec 29 16:16:11 GMT 2023
;; MSG SIZE  rcvd: 282
```

Comment (billywhizz) [reply cli/tests/unit/net_test.ts:435]: i also see the same thing here. sounds like something might be temporarily broken on google side?

https://dns.google/query?name=8.8.8.8.in-addr.arpa

**Comment by billywhizz** [comment]
if there is anything you need me to do to get this landed please shout. maybe it has been fixed elsewhere?

**Comment by littledivy** [comment]
@billywhizz Sorry for the late reviews. This looks good, can you rebase?

**Review by dsherret — APPROVED** [review]
LGTM. Thanks, Andrew!

**Pull request merged**